### ISSUE TITLE
Datepicker formats incorrect

### ISSUE THREAD

**Issue by WrDX** [Dogfalo/materialize#5516]
## Expected Behavior
```{format: 'yyyy-mm-dd'}```

I would expect '2018-01-01'.

## Current Behavior

It results in '2018-1-1' (what I would expect from a 'yyyy-m-d', not 'yyyy-mm-dd')

## Possible Solution

Follow formatting rules from datepicker.js:
http://amsul.ca/pickadate.js/date/#formatting-rules

**Comment by DanielRuf** [comment]
https://github.com/Dogfalo/materialize/blob/master/js/date_picker/picker.date.js#L689

@Dogfalo the `_.lead()` statement was not completely refactored as it seems.

**Comment by Dogfalo** [comment]
Please follow the issue guidelines and state the version you are using.

**Comment by DanielRuf** [comment]
@Dogfalo it is in the current v1-dev.

**Comment by Dogfalo** [comment]
The file you linked is on master not v1-dev

**Comment by WrDX** [comment]
I have the issue on 1.0.0-alpha3 (using cloudflare CDN)

**Comment by DanielRuf** [comment]
I know @Dogfalo, I just wanted to show you what is missing. The _.lead().

**Comment by DanielRuf** [comment]
See https://github.com/Dogfalo/materialize/blob/v1-dev/js/datepicker.js#L657-L684

**Comment by DanielRuf** [comment]
https://stackoverflow.com/questions/6040515/how-do-i-get-month-and-date-of-javascript-in-2-digit-format

**Comment by Nrde** [comment]
So is this going to be fixed?
I had to use this ugly workaround.
```
Options = {
     format: 'yyyy-mm-dd', /* this does not work */
     onClose: function() {
        this.el.value = this.date.getFullYear()+"-"+("0" + (this.date.getMonth() + 1)).slice(-2)+"-"+("0" + (this.date.getDate())).slice(-2);
    }

};```

**Comment by DanielRuf** [comment]
I might provide a PR if no other contributor has the time for it.
Basically I'll reimplement `_.lead()`.

**Comment by ray007** [comment]
There currently are not formats for `d` and `m` in datepicker.
Also: the format for `yy` throws an exception.
I'll try to find time to make a PR later today if you do not beat me to it.

**Comment by ray007** [comment]
push request submitted (#5573), again with minimal changes necessary.

**Comment by Dogfalo** [comment]
Merged #5573
